src/otaclient/_types.py

```diff
@@ -124,13 +125,46 @@ class OTAClientStatus:
 
 
 @dataclass
-class UpdateRequestV2:
+class MultipleECUStatusFlags:
+    any_in_update: mp_sync.Event
+    any_requires_network: mp_sync.Event
+    all_success: mp_sync.Event
+
+
+#
+# ------ OTA requests IPC ------ #
+#
+
+
+class IPCResEnum(StrEnum):
+    ACCEPT = "ACCEPT"
+    REJECT_BUSY = "REJECT_BUSY"
+    """The request has been rejected due to otaclient is busy."""
+    REJECT_OTHER = "REJECT_OTHER"
+    """The request has been rejected for other reason."""
```

Comment:
> grpc for IPC(between api/core) will be more maintainable.
That is what I call "coupling with specific OTA Service API version", and I want to avoid it.
The core idea is, even otaclient itself doesn't have grpc server implementation or OTA Service API definition, the ota_core itself should be able to run.

Another thing is, the current OTA Service API definition lack of precious enum definition relating to request accepted or rejected.

Reply:
The ultimate objective of the otaclient re-architecture is to make the ota_core part of otaclient(boot_control, create_standby, ota_core, etc.) become standalone, instead of implementing it as a service handler of grpc server.

We can easily integrate the new ota_core behind any front-end interface, no mater it is a grpc API interface, or even a simple CLI interface.

Reply:
Thank you, understood.

> Another thing is, the current OTA Service API definition lack of precious enum definition relating to request accepted or rejected.

I think this is important point.  Currently, `ota-core` supports a few return response code and messages. When field engineer fails to ota update, they will hope to know the precise reason and how to recover it without asking engineers / checking cloudwatch log.
In future, we will need to support much more error code. So I'm considering what kind of interface(`ota-core` response) will be scalable.

Reply:
Yes the original API design even doesn't have a msg field in the response msg, I feel very confused about it 👼 
We definitely need to define a new API with more consideration.

Reply:
Ah and another important point of using internal interface is that we can keep ota_core unchanged when we introduce new OTA Service API, which will definitely happen in the near future.